### PR TITLE
xdp: sessionOffload XDP progress

### PR DESCRIPTION
**Pull request by mestery**
This commit adds in a simple eBPF/XDP program which will handle the backend of the sessionOffload API. For now, we simply define the eBPF map used for communication from the golang server and the eBPF/XDP program. The XDP program skeleton program is taken from a simple example program I've written in the past [1]. It will be fleshed out to actually operate on the sessions in the eBPF Map.

Related to #68

[1] https://github.com/mestery/xdp-skeleton

Signed-off-by: Kyle Mestery <mestery@mestery.com>

### CODE REVIEW

File: .github/workflows/framework-golang.yml [hunk 1]
```diff
@@ -54,6 +54,20 @@ jobs:
         images: ${{ env.REGISTRY }}/${{ github.repository_owner }}/opi-sessionoffload-${{ matrix.package }}
 
     - name: Build and push Docker image
+      if: package != 'xdp'
```
Comment (glimchb) [inline .github/workflows/framework-golang.yml:57]: should be `${{ matrix.package }}` ?

**Review by glimchb — COMMENTED** [review]
i'm lost in review... this is too big for me... can we review in pieces ?

File: .github/workflows/framework-golang.yml [hunk 2]
```diff
@@ -65,6 +66,20 @@ jobs:
         cache-from: type=gha
         cache-to: type=gha,mode=max
 
+    - name: Build and push Docker image
+      if: matrix.package == 'xdp'
+      id: build-and-push-xdp
+      uses: docker/build-push-action@ffe65bfc276d2ef003d5ea483a9947a8f6d408da
+      with:
+        context: openoffload/go
+        file: openoffload/go/${{ matrix.package }}/Dockerfile
```
[the remark this answers is not in the document]
Comment (mestery) [reply .github/workflows/framework-golang.yml:75]: No, it's also the build location (xdp builds one level up).

**Review by glimchb — APPROVED** [review]
lgtm

**Pull request merged**